src/Service/S3/src/Signer/SignerV4ForS3.php

```diff
@@ -115,8 +115,15 @@ protected function convertBodyToStream(SigningContext $context): void
             return;
         }
 
+        // Add content-encoding for chunked stream if available
+        $customEncoding = $context->getRequest()->getHeader('content-encoding');
+        $contentEncoding = 'aws-chunked';
+        if ($customEncoding) {
+            $contentEncoding .= sprintf(', %s', $customEncoding);
+        }
```

Comment:
`sprintf` looks overkill to me here, it can be replaced by `', '. $customEncoding` or string interpolation?

Or even remove the entire if.
```suggestion
        $contentEncoding = $customEncoding ? "aws-chunked, $customEncoding" : 'aws-chunked';
```